### DISCUSSION TITLE
[Object detection & Keypoint regression] Detectron2 VS MMDetection VS MMPose

### DISCUSSION THREAD

**Discussion by joihn** [facebookresearch/detectron2#3539] [Q&A] [answered]
The project I'm working on involve single class object detection and single keypoint detection (onto the object).
I have difficulties picking the best framework:
- **Detectron2** seems easy to use and does both bounding box and keypoints, but the model zoo seems small.
- **MMDetection** model zoo seems very vast with SOTA, but does not offer keypoint detection it seems.
- **MMPose** seems to does keypoint regression, but only for human, and the outputed BoundingBox (important for me) might not be accurate since the main goal is only pose detection 

Anyone has some tipps on which framework to choose ?

**Comment by sukjunhwang** [comment, marked as answer]
I believe it depends on your preference.
With numerous contributors to the codebases of MM, MM provides many handy functions and codes of SOTA models.
So, you might find MM more configurable and easy to customize.

On the other hand, in my opinion, the codes of Detectron2 are much neat and easy to follow. (maybe it's because I am used to Detectron2)
So even though it might take some time, I prefer to reproduce models into Detectron2.
